compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1587,11 +1587,29 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                         let o = self.resolve_vars_if_possible(o);
                         if !self.predicate_may_hold(&o) {
                             result = ProbeResult::NoMatch;
-                            possibly_unsatisfied_predicates.push((
-                                o.predicate,
-                                None,
-                                Some(o.cause),
-                            ));
+                            let parent_o = o.clone();
+                            let implied_obligations =
+                                traits::elaborate_obligations(self.tcx, vec![o]);
+                            for o in implied_obligations {
+                                let parent = if o == parent_o {
+                                    None
+                                } else {
+                                    if o.predicate.to_opt_poly_trait_pred().map(|p| p.def_id())
+                                        == self.tcx.lang_items().sized_trait()
```

Comment:
```suggestion
                                    if let Some(trait_pred) = o.predicate.to_opt_poly_trait_pred()
                                        && self.tcx.lang_items().sized_trait() == Some(trait_pred.def_id())
```

This will never happen in practice, but it's better to deal with the `None == None` case when lang-items are not defined.

Reply:
What is the `None == None` case you envision? Because `trait_pred` might have been `None` (so we don't get into this logic) and `sized_trait()` might be `None` which might cause a build where the sized trait lang item isn't present to be too verbose (with extra labels), no other behavioral change.